### DISCUSSION TITLE
[Vue] why is this incompatible with vue 2.0 ?

### DISCUSSION THREAD

**Discussion by aon108** [tailwindlabs/headlessui#417] [Help]
I need to get this running using vue 2.0 as my platform hasnt been upgraded to 3.0. 
What parts specifically rely on vue 3.0 api?
Cheers

**Comment by aon108** [comment]
looking through the source code a number of breaking changes due to composition-api; like setup() for example. Will just have to eagerly wait for migration to Vue3

**Comment by adamwathan** [comment]
The bigger issue than the composition API honestly is the massive breaking changes to render functions and the VNode format:

https://v3.vuejs.org/guide/migration/render-function-api.html#_3-x-syntax-4

Our entire library is written with render functions since it's headless, and it's very different between Vue 2 and Vue 3.
